projects/schematics/src/migrations/2_0/validate.ts

```diff
@@ -0,0 +1,29 @@
+import {
+  Rule,
+  SchematicContext,
+  SchematicsException,
+  Tree,
+} from '@angular-devkit/schematics';
+import {
+  getAngularVersion,
+  getMajorVersionNumber,
+} from '../../shared/utils/package-utils';
+
+export function validate(): Rule {
+  return (tree: Tree, _context: SchematicContext) => {
+    const angularVersion = getAngularVersion(tree);
+    validateAngularVersion(angularVersion);
+
+    return tree;
+  };
+}
+
+export function validateAngularVersion(angularVersion: string): void {
```

Comment:
Maybe we can go with something more generic and move it to shared/utils?

Reply:
Not sure if this should be a generic method. We could always do that of course, but I don't see it being use in multiple places for now, besides this validation. Do you have some specific use case on mind?

Reply:
Tried to come up with some example, but maybe I overreacted with this one :) Let's leave it as is.

Reply:
It's a valid proposition :)